lib/config/validation.spec.ts

```diff
@@ -1075,5 +1022,271 @@ describe('config/validation', () => {
+          true,
+          config,
+        );
+        expect(warnings).toEqual([
+          {
+            message: 'Configuration option `baseDir` should be a string.',
+            topic: 'Configuration Error',
+          },
+        ]);
+      });
+
+      it('requireConfig', async () => {
+        const config = {
+          requireConfig: 'invalid' as never,
+        };
+        const { warnings } = await configValidation.validateConfig(
+          true,
+          config,
+        );
+        expect(warnings).toEqual([
+          {
+            message:
+              'Invalid value `invalid` for `requireConfig`. The allowed values are required, optional, ignored.',
+            topic: 'Configuration Error',
+          },
+        ]);
+      });
+
+      it('dryRun', async () => {
+        const config = {
+          dryRun: 'invalid' as never,
+        };
+        const { warnings } = await configValidation.validateConfig(
+          true,
+          config,
+        );
+        expect(warnings).toEqual([
+          {
+            message:
+              'Invalid value `invalid` for `dryRun`. The allowed values are extract, lookup, full.',
+            topic: 'Configuration Error',
+          },
+        ]);
+      });
+
+      it('repositoryCache', async () => {
+        const config = {
+          repositoryCache: 'invalid' as never,
+        };
+        const { warnings } = await configValidation.validateConfig(
+          true,
+          config,
+        );
+        expect(warnings).toEqual([
+          {
+            message:
+              'Invalid value `invalid` for `repositoryCache`. The allowed values are enabled, disabled, reset.',
+            topic: 'Configuration Error',
+          },
+        ]);
+      });
+
+      it('onboardingConfigFileName', async () => {
+        const config = {
+          onboardingConfigFileName: 'invalid' as never,
+        };
+        const { warnings } = await configValidation.validateConfig(
+          true,
+          config,
+        );
+        expect(warnings).toEqual([
+          {
+            message: `Invalid value \`invalid\` for \`onboardingConfigFileName\`. The allowed values are ${configFileNames.join(', ')}.`,
+            topic: 'Configuration Error',
+          },
+        ]);
+      });
+
+      it('gitUrl', async () => {
+        const config = {
+          gitUrl: 'invalid' as never,
+        };
+        const { warnings } = await configValidation.validateConfig(
+          true,
+          config,
+        );
+        expect(warnings).toEqual([
+          {
+            message:
+              'Invalid value `invalid` for `gitUrl`. The allowed values are default, ssh, endpoint.',
+            topic: 'Configuration Error',
+          },
+        ]);
+      });
+    });
+
+    it('validates boolean type options', async () => {
+      const config = {
+        unicodeEmoji: false,
+        detectGlobalManagerConfig: 'invalid-type' as never,
```

Comment:
use ts-expect-error comment instead of never

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments